akka-stream-tests/src/test/scala/akka/stream/scaladsl/RestartSpec.scala

```diff
@@ -654,10 +653,12 @@ class RestartSpec
                   case other      => other
                 }
                 .to(Sink
-                  .foreach(flowInSource.sendNext)
+                  .foreach[String] { msg =>
+                    probe.ref ! msg
+                  }
                   .mapMaterializedValue(_.onComplete {
-                    case Success(_) => flowInSource.sendNext("in complete")
-                    case Failure(_) => flowInSource.sendNext("in error")
```

Comment:
Just as a reminder of the problem: since `flowInSource` was a stream probe, we can't just imperatively push to it from two different threads, what was failing was that it had not yet seen demand (from the `buffer(4)`) in one of the `sendNext` cases and would throw, leading to a subsequent expectation fail in the actual test.